cluster-api-gcp/CONTRIBUTING.md

```diff
@@ -0,0 +1,59 @@
+# Contributing Guidelines
+
+## Build
+
+```bash
+$ cd $GOPATH/src/k8s.io/
+$ git clone git@github.com:kubernetes/kube-deploy.git
+$ cd kube-deploy/cluster-api-gcp/
+$ go build
+```
+
+This will create a binary `cluster-api-gcp` that you can use to manage a GCP cluster.
+
+## Developing
+
+When making changes to the machine controller, it's generally a good idea to delete any existing cluster created with an older version of the cluster-api.
+
+```bash
+$ ./cluster-api-gcp delete
+```
+
+After making changes to the machine controller or the actuator, you need to follow these two steps:
+
+1. Rebuild the machine-controller image.
+
+	```bash
+	$ cd machine-controller
+	$ make push fix-image-permissions
```

Comment:
Are you sure this is intentional to explicitly fix image permissions?

Reply:
In my dev process, I found that fixing permissions was needed at least the first time I pushed the image. I don't think it hurts to run it on every push.